_episodes/09-github.md

```diff
@@ -93,18 +89,184 @@ We can check that the command has worked by running `git remote -v`:
+
+~~~
+$ ssh-keygen -t ed25519 -C "vlad@tran.sylvan.ia"
+~~~
+{: .language-bash}
+
+If you are using a legacy system that doesn't support the Ed25519 algorithm, use:
+`$ ssh-keygen -t rsa -b 4096 -C "your_email@example.com"`
+
+~~~
+Generating public/private ed25519 key pair.
+Enter file in which to save the key (/c/Users/Vlad Dracula/.ssh/id_ed25519):
+~~~
+{: .output}
+
+We want to use the default file, so just press <kbd>Enter</kbd>.
+
+~~~
+Created directory '/c/Users/Vlad Dracula/.ssh'.
+Enter passphrase (empty for no passphrase):
+~~~
+{: .output}
+
+Now, it is prompting Dracula for a passphrase.  Since he is using his lab’s laptop that other people sometimes have access to, he wants to create a passphrase.  Be sure to use something memorable or save your passphrase somewhere, as there is no "reset my password" option. 
+
+~~~
+Enter same passphrase again:
+~~~
+{: .output}
+
+After entering the same passphrase a second time, we receive the confirmation
+
+~~~
+Your identification has been saved in /c/Users/Vlad Dracula/.ssh/id_ed25519
+Your public key has been saved in /c/Users/Vlad Dracula/.ssh/id_ed25519.pub
+The key fingerprint is:
+SHA256:SMSPIStNyA00KPxuYu94KpZgRAYjgt9g4BA4kFy3g1o vlad@tran.sylvan.ia
+The key's randomart image is:
++--[ED25519 256]--+
+|^B== o.          |
+|%*=.*.+          |
+|+=.E =.+         |
+| .=.+.o..        |
+|....  . S        |
+|.+ o             |
+|+ =              |
+|.o.o             |
+|oo+.             |
++----[SHA256]-----+
+~~~
+{: .output}
+
+The "identification" is actually the private key. You should never share it.  The public key is appropriately named.  The "key fingerprint" 
+is a shorter version of a public key.
+
+Now that we have generated the SSH keys, we will find the SSH files when we check.
+
+~~~
+ls -al ~/.ssh
+~~~
+{: .language-bash}
+
+~~~
+drwxr-xr-x 1 Vlad Dracula 197121   0 Jul 16 14:48 ./
+drwxr-xr-x 1 Vlad Dracula 197121   0 Jul 16 14:48 ../
+-rw-r--r-- 1 Vlad Dracula 197121 419 Jul 16 14:48 id_ed25519
+-rw-r--r-- 1 Vlad Dracula 197121 106 Jul 16 14:48 id_ed25519.pub
+~~~
+{: .output}
+
+### Copy the public key to GitHub
+Now we have a SSH key pair and we can run this command to check if GitHub can read our authentication.  
+
+~~~
+ssh -T git@github.com
+~~~
+{: .language-bash}
+
+
+~~~
+The authenticity of host 'github.com (192.30.255.112)' can't be established.
+RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
+This key is not known by any other names
+Are you sure you want to continue connecting (yes/no/[fingerprint])? y
+Please type 'yes', 'no' or the fingerprint: yes
+Warning: Permanently added 'github.com' (RSA) to the list of known hosts.
+git@github.com: Permission denied (publickey).
+~~~
+{: .output}
+
+Right, we forgot that we need to give GitHub our public key!  
+
+First, we need to copy the public key.  Be sure to include the `.pub` at the end, otherwise you’re looking at the private key. 
+
+~~~
+cat ~/.ssh/id_ed25519.pub
```

Comment:
maybe pbcopy instead?  Not sure if that works on all systems though so might have to include both.

Reply:
pbcopy does not exist on Linux. The equivalent is `xclip -selection clipboard`.

Reply:
I will add something for xclip.

I taught this last week and copying text from and pasting to the terminal was confusing, even though I planned for it. Using xclip seems safest and most universal.

Reply:
Alas, neither of these are universal; xclip doesn't exit on mac. The way I've gotten it to work cross-platform was to pipe stderr to /dev/null and use the logical OR to fallback when the first command was not detected:

https://github.com/zkamvar/config-files/blob/da7b5668c5c0d8dd801f2684df6e3ddff8086989/.bash_funs#L7

and I would not recommend this for beginners because of all the extra concepts

Reply:
[This page](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account) on GitHub's instructions uses `clip`:

```
$ clip < ~/.ssh/id_ed25519.pub
# Copies the contents of the id_ed25519.pub file to your clipboard
```

Would this be cross-platform?

Reply:
Honestly, I think this is just more reason to include it in the setup instructions since we can more easily specify platform-specific instructions.

Reply:
I made a little callout called "Copy and paste in the unix shell" on the setup lesson. Since I taught this to people with little/no experience with unix shell, this episode required a bit of intro to shell anyway.

Interested in feedback! Some thoughts:

1. While a CLI tool like xcopy keeps them in the shell, their next step is to paste the text in their browser anyway. Most learners are familiar with Copy and Paste and appreciate the heads-up about their function in the shell.
2. It can be glossed over if this is part of a SWC workshop where shell is already discussed. That being said, I couldn't find where copy and paste are mentioned. Think that's intentional?